commons/protocol/src/main/scala/sbt/protocol/Keys.scala

```diff
@@ -1,80 +1,29 @@
 package sbt.protocol
 
 import java.net.URI
-import ScalaShims.ManifestFactory
-import play.api.libs.json._
-
-import Reads._
-import Writes._
-import sbt.GenericSerializers._
-
-/**
- *  Represents the type information we can serialize over a network
- *  from sbt.  We try to preserve, as much as possible, the items inside
- *  a scala.reflect.Manfiest.
- */
-final case class TypeInfo(erasureClass: String, typeArguments: Seq[TypeInfo] = Seq.empty) {
-  override def toString = erasureClass + (
-    if (typeArguments.isEmpty) ""
-    else typeArguments.mkString("[", ",", "]"))
-
-  def toManifest(cl: ClassLoader = TypeInfo.getClass.getClassLoader): Option[Manifest[_]] = {
-    val args = typeArguments.map(_.toManifest(cl))
-    if (args.forall(_.isDefined)) {
-      val realArgs = args.map(_.get)
-      // Now we look at our class....
-      erasureClass match {
-        // TODO - Special casing classes...
-        case "boolean" => Some(ManifestFactory.Boolean)
-        case "short" => Some(ManifestFactory.Short)
-        case "int" => Some(ManifestFactory.Int)
-        case "long" => Some(ManifestFactory.Long)
-        case "float" => Some(ManifestFactory.Float)
-        case "double" => Some(ManifestFactory.Double)
-        case "Unit" => Some(ManifestFactory.Unit)
-        case default => try {
-          val ourClass = cl.loadClass(erasureClass)
-          val mf =
-            if (realArgs.isEmpty) {
-              ManifestFactory.classType(ourClass)
-            } else {
-              ManifestFactory.classType(ourClass, realArgs.head, realArgs.tail: _*)
-            }
-          Some(mf.asInstanceOf[Manifest[_]])
-        } catch {
-          case _: ClassNotFoundException =>
-            None
-        }
-      }
-    } else None
-  }
-}
-object TypeInfo {
-  def fromManifest[T](mf: Manifest[T]): TypeInfo = {
-    TypeInfo(
-      mf.erasure.getName,
-      mf.typeArguments map (x => fromManifest(x)))
-  }
-
-  def fromClass[T](klass: Class[T]): TypeInfo = {
-    TypeInfo(klass.getName, Nil)
-  }
-
-  implicit val reads: Reads[TypeInfo] = Json.reads[TypeInfo]
-  implicit val writes: Writes[TypeInfo] = Json.writes[TypeInfo]
-}
+import sbt.serialization._
+import sbt.serialization.functions._
+import scala.pickling.{ SPickler, Unpickler }
+import scala.pickling.internal.AppliedType
 
 /**
  * This represents a "key" in sbt.
  *  Keys have names and "types" associated.
+ * TODO FIXME it is bad to have an internal type from pickling here
```

Comment:
#240 got merged to pickling2. This branch probably split off from it prior to the merge?